doc/ot_api_doc.h

```diff
@@ -142,6 +142,7 @@
  * @defgroup api-mesh-diag            Mesh Diagnostics
  * @defgroup api-ncp                  Network Co-Processor
  * @defgroup api-network-time         Network Time Synchronization
+ * @defgroup api-radio-stats          Radio Statistics
```

Comment:
LGTM.
One suggestion: How about if we rename and use `api-radio`? So it is more generic and can potentially be used for any future `otRadio` APIs?

Reply:
I'm fine with that, @gavinmcgovern if you can update the source radio-stats.h file to align with that as well